### PR TITLE
Add tests

### PR DESCRIPTION
**Pull request by pburtchaell**
- [x] Dispatch actions for fulfilled promises
- [x] Dispatch actions for rejected promises
- [x] Test standard actions without promise
- [ ] Test async action creators, e.g., the broadcast meta property 
- [ ] Test that meta and error are not included if they are not passed in

### CODE REVIEW

**Comment by pburtchaell** [comment]
@tomatau Here is my initial work for the tests. Let me know what you think.

File: test/index.spec.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    const mockStore = configureStore(middlewares);
+    const mockActionWithoutPromise = { type: 'GET_POST' };
+    const mockRejectedAction = {
+      type: 'GET_POST',
+      payload: {
+        promise: () => {
+          return new Promise((resolve, reject) => {
+            reject(new Error());
+          });
+        }
```
Comment (tomatau) [inline test/index.spec.js]: These two can be for a little less boilerplate

```js
    const mockRejectedAction = {
      type: 'GET_POST',
      payload: {
        promise: () => Promise.reject(new Error())
      }
    };
    const mockFulfilledAction = {
      type: 'GET_POST',
      payload: {
        promise: () => Promise.resolve({})
      }
    };
```

File: test/index.spec.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+      }
+    };
+
+    it('must pass action to next if not a promise', done => {
+      const store = mockStore({}, [mockActionWithoutPromise], done);
+      store.dispatch(mockActionWithoutPromise);
+    });
+
+    it('must dispatch rejected action if promise is rejected', done => {
+      const store = mockStore({}, [mockRejectedAction], done);
```
Comment (tomatau) [inline test/index.spec.js]: I don't understand this, it's testing that the action is dispatched with exactly the same type and also including the same payload, shouldn't we be testing that it's dispatched our three suffix types?

```
    const first = {
      type: 'GET_POST_PENDING'
    };
    const rejected = {
      type: 'GET_POST_REJECTED',
      error: true
    };
    const fulfilled = {
      type: 'GET_POST_FULFILLED',
      payload: {}
    };
```

File: test/index.spec.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+      const actionHandler = nextHandler();
+      chai.assert.isFunction(actionHandler);
+    });
+  });
+
+  describe('handle action', () => {
+    const middlewares = [promiseMiddleware()];
+    const mockStore = configureStore(middlewares);
+    const mockActionWithoutPromise = { type: 'GET_POST' };
+    const mockRejectedAction = {
```
Comment (tomatau) [inline test/index.spec.js]: these mock actions aren't returning promise instances, they're returning functions which themselves return promises

**Comment by tomatau** [comment]
I like the start of this and you're definitely on the right lines, but the tests didn't make some sense to me.

I don't think we need to write tests for actions with `payload.promise` being a function that returns a promise, the `payload.promise` should be a promise itself. Then we can test that our `_PENDING`, + `_FULFILLED` or `_REJECTED` are applied appropriately.

**Comment by pburtchaell** [comment]
Thanks for looking this over. I'm going to put it on my todo list to review this again this week.

**Comment by pburtchaell** [comment]
So I merged everything in and rebased to master. Looks like the changes made in the last version are now causing errors to fail. Error:

```
    var dispatch = _ref.dispatch;
                       ^

TypeError: Cannot read property 'dispatch' of undefined
```

**Comment by tomatau** [comment]
Line 6 of tests, the middleware is being double invoked so it isn't being passed through the configure store API.
```
  const nextHandler = promiseMiddleware()();
```

As it's expected to be passed into redux's create store, maybe we should change the above line to simulate that.

The other tests where it's being passed through to mockStore should still work.

**Pull request merged**